Automattic/pocket-casts-ios
# Transcripts: HTML transcripts content layout

| 📘 Part of: #1848  |
|:---:|

This PR adds the formatting of transcripts that are provided in the `plain/html` format.

It does the following:

- It removes all HTML tags
- Adds new lines wherever there is HTML `BR`
- Removes speaker information
- Removes sound descriptors. Ex: [laughs] [music]
- Removes any extra spaces between lines and words



## To test

- Start the app
- Ensure that you have the `transcripts` and `newShowNotesEndpoint` FF enabled
- Start playing an episode with transcript support using only the `plain/html` format. Ex: [If Books could Kill](https://pca.st/v78v6vve) or [Maintenance Phase](https://pca.st/97b36s90)
- Open the full screen player
- Tap on the transcript shelf button 
- See that the layout of text, is respecting paragraphs and no "Speaker" entries or other xml kind of tags `<>` are present on on the text.

## Checklist
